src/vm/dllimport.cpp

```diff
@@ -6355,22 +6374,89 @@ HINSTANCE NDirect::LoadLibraryModule(NDirectMethodDesc * pMD, LoadLibErrorTracke
                     flags |= dllImportSearchPathFlag;
                 }
 
-                hmod = LocalLoadLibraryHelper(currLibNameVariation, flags, pErrorTracker);
+                hmod = LocalLoadLibraryHelper(currLibNameVariation, flags, &pErrorTracker);
             }
-            else if (searchAssemblyDirectory)
+            else if (searchAssemblyDirectory && pAssembly != NULL)
             {
-                Assembly* pAssembly = pMD->GetMethodTable()->GetAssembly();
-                hmod = LoadFromPInvokeAssemblyDirectory(pAssembly, currLibNameVariation, loadWithAlteredPathFlags | dllImportSearchPathFlag, pErrorTracker);
+                hmod = LoadFromPInvokeAssemblyDirectory(pAssembly, currLibNameVariation, loadWithAlteredPathFlags | dllImportSearchPathFlag, &pErrorTracker);
             }
         }
 
         // This call searches the application directory instead of the location for the library.
         if (hmod == NULL)
         {
-            hmod = LocalLoadLibraryHelper(currLibNameVariation, dllImportSearchPathFlag, pErrorTracker);
+            hmod = LocalLoadLibraryHelper(currLibNameVariation, dllImportSearchPathFlag, &pErrorTracker);
         }
     }
 
+    return hmod.Extract();
+}
+
+HINSTANCE NDirect::LoadLibraryModule(NDirectMethodDesc * pMD)
+{
+    LoadLibErrorTracker pErrorTracker;
```

Comment:
This is an annoying style that is confusing. The `LoadLibErrorTracker` type appears to be a struct/class - unless it is `typedef`'d as a pointer and that would be evil, but the variable name uses the hungarian prefix for pointer (i.e. `p`). I would remove that prefix.

Reply:
same above